### PR TITLE
FACES-2697 jsp

### PR DESCRIPTION
**Pull request by juangon**
Hi @ngriffin7a. I think I came across a good solution. I am using JspFactory (J2EE standard javax.servlet) for obtaining the PageContext, so we don't have to worry about the implementation.

I followed similar approach as in some parts in Liferay.

Additionally, I've removed some classes as they seems to not be needed anymore.

Please, backport it to 2.x.

Thanks!

### CODE REVIEW

**Comment by juangon** [comment]
BTW I've done tests in Tomcat and Wildfly in LF7, and everyting seems to still work.

File: src/main/java/com/liferay/faces/util/jsp/internal/PageAdapter.java [hunk 1]
```diff
@@ -16,23 +16,28 @@
...
+public class PageAdapter extends HttpServlet {
 
 	private ServletConfig servletConfig;
+	private ServletContext servletContext;
 
-	public PageAdapter(ServletConfig servletConfig) {
-		this.servletConfig = servletConfig;
+	public PageAdapter(ServletContext servletContext) {
+		this.servletConfig = new PageAdapterServletConfig();
+		this.servletContext = servletContext;
```
Comment (juangon) [inline src/main/java/com/liferay/faces/util/jsp/internal/PageAdapter.java]: I've adapted this class for the needs in this issue. I've been searching and I couldn't find the place where it's used.

**Comment by stiemannkj1** [comment]
@juangon, these changes are great!

@ngriffin7a, since Juan has simplified this so much, I'm wondering if we could eliminate this API completely. I'm going to do some testing, but I think it might be possible.

**Comment by ngriffin7a** [comment]
@juangon Due to a small merge conflict caused by https://issues.liferay.com/browse/FACES-2841 I will close this pull request. After the release, please send a new pull request for https://issues.liferay.com/browse/FACES-2844 Thanks!

**Pull request closed without merging**